src/bindgen/parser.rs

```diff
@@ -84,6 +85,134 @@ pub(crate) fn parse_lib(lib: Cargo, config: &Config) -> ParseResult {
     Ok(context.out)
 }
 
+/// Parses a Rust MIR source file
+pub fn parse_mir(mir_path: &FilePath, config: &Config, parse: &mut Parse) -> Result<(), Error> {
+
+    let mod_name = mir_path.file_stem().unwrap().to_str().unwrap();
+    let mut mir_file = File::open(mir_path).map_err(|_| Error::ParseCannotOpenFile {
+         crate_name: mod_name.to_owned(), src_path: mir_path.to_str().unwrap().to_owned()
+        }
+    )?;
+
+    // Struct to track presence of associated types
+    struct Assoc{
+        ret: bool,
+        args: Vec<usize>,
+    }
+
+    // Check each function signature for associated types
+    for func in parse.functions.iter_mut(){
+
+        // Parse function to populate struct
+        let mut assoc_track = Assoc{
+            ret: false,
+            args: Vec::new(),
+        };
+        if let Type::Path(GenericPath{qself: Some(_), ..}) = &func.ret{
+            assoc_track.ret = true;
+        }
+        for index in 0..func.args.len(){
+            if let Type::Path(GenericPath{qself: Some(_), ..}) = func.args[index].ty{
+                assoc_track.args.push(index);
+            }
+        }
+
+        if !assoc_track.ret && assoc_track.args.is_empty(){
+           continue;
+        }
+
+        let pat = format!(r"fn[^{{]*{}[^{{]*", func.path.name());
+        let re = Regex::new(&pat).unwrap();
```

Comment:
regex is no longer necessary since you are using `syn::parse_str` to produce `syn::Signature`. If `syn::parse_str` fails just continue to the next string